modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/components/ActionCreationForm.tsx

```diff
@@ -0,0 +1,419 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayButton from '@clayui/button';
+import ClayForm, {ClayInput, ClaySelectWithOption} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import ClayLayout from '@clayui/layout';
+import ClayPanel from '@clayui/panel';
+import {InputLocalized} from 'frontend-js-components-web';
+import {fetch} from 'frontend-js-web';
+import React, {useEffect, useState} from 'react';
+
+import {API_URL, OBJECT_RELATIONSHIP} from '../Constants';
+import {FDSViewType} from '../FDSViews';
+import RequiredMark from '../components/RequiredMark';
+import openDefaultFailureToast from '../utils/openDefaultFailureToast';
+import openDefaultSuccessToast from '../utils/openDefaultSuccessToast';
+
+const MESSAGE_TYPES = [
+	{
+		label: Liferay.Language.get('info'),
+		value: 'info',
+	},
+	{
+		label: Liferay.Language.get('secondary'),
+		value: 'secondary',
+	},
+	{
+		label: Liferay.Language.get('success'),
+		value: 'success',
+	},
+	{
+		label: Liferay.Language.get('danger'),
+		value: 'danger',
+	},
+	{
+		label: Liferay.Language.get('warning'),
+		value: 'warning',
+	},
+];
+
+const TYPES = [
+	{
+		label: Liferay.Language.get('link'),
+		value: 'link',
+	},
+];
+
+interface IFDSActionCreationFormInterface {
```

Comment:
nit: no need for `*Interface` at the end of the name, you already prefix with `I` so that should be enough. You can use `*Props` though since it is for a component props